app/controllers/SelectPaginatedClientsController.scala

```diff
@@ -143,12 +143,7 @@ class SelectPaginatedClientsController @Inject()(
                 .flatMap(_ => {
                   if (formData.submit == CONTINUE_BUTTON) {
                     // check selected clients from session cache AFTER saving (removed de-selections)
-                    val hasSelectedClients = for {
-                      selectedClientIds <- sessionCacheService.get(SELECTED_CLIENT_IDS)
-                      // if "empty" returns Some(Vector()) so .nonEmpty on it's own returns true
-                    } yield selectedClientIds.getOrElse(Seq.empty).nonEmpty
-                    hasSelectedClients.flatMap(selectedNotEmpty => {
-                      if (selectedNotEmpty) {
+                    if (maybeSelected.getOrElse(Seq.empty).nonEmpty) {
```

Comment:
still need to get SELECTED_CLIENTS after `.savePageOfClients(formData)` incase all have been deselected

Reply:
good point

Reply:
I think we'll never hit this else case at the moment
`} else { // render page with empty client error`

since this is now the same as line 121
if (maybeSelected.getOrElse(Seq.empty).nonEmpty) {

if it's empty, the form won't bind and it will be treated as `formWithErrors`